### PR TITLE
feat: Introduce stats task for import

### PR DESCRIPTION
**Pull request by bigsheeper**
This PR introduce stats task for import:
1. Define new `Stats` and `IndexBuilding` states for importJob
2. Add new stats step to the import process: trigger the stats task and wait for its completion
3. Abort stats task if import job failed

issue: https://github.com/milvus-io/milvus/issues/33744

### CODE REVIEW

**Comment by bigsheeper** [comment]
/hold
wait for https://github.com/milvus-io/milvus/pull/35054 merged

File: internal/datacoord/garbage_collector.go [hunk 1]
```diff
@@ -423,7 +423,7 @@ func (gc *garbageCollector) recycleDroppedSegments(ctx context.Context) {
 	for _, segment := range all {
 		cloned := segment.Clone()
 		binlog.DecompressBinLogs(cloned.SegmentInfo)
-		if cloned.GetState() == commonpb.SegmentState_Dropped {
+		if cloned.GetState() == commonpb.SegmentState_Dropped && !cloned.GetIsImporting() {
```
Comment (czs007) [inline internal/datacoord/garbage_collector.go:426]: why?

**Pull request merged**